l10n_eu_product_adr/models/product_template.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019 Iryna Vyshnevska (Camptocamp)
+# Copyright 2021 Opener B.V. <stefan@opener.amsterdam>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    adr_goods_on_variants = fields.Boolean(
+        compute="_compute_adr_goods_on_variants",
+        help="Indicates whether the adr configuration is different for each variant.",
+    )
+    is_dangerous = fields.Boolean(
+        related="product_variant_ids.is_dangerous",
```

Comment:
That seemed to me to be a necessary consequence of @mmequignon's request to allow different settings per product variant. If it is not allowed, it is not possible to extend the number of variants by adding product attribute options on the template because the actual dangerous good is not available to the Odoo mechanism that expands the number of variants.